.github/scripts/variables_documentation.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright 2023 Efabless Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+
+documented_elsewhere = """
+CREATE_REPRODUCIBLE_FROM_SCRIPT
+"""
+internal_variables = """
+CONFIGS
+CORE_HEIGHT
+CORE_WIDTH
+CURRENT_DEF
+CURRENT_GDS
+CURRENT_GUIDE
+CURRENT_INDEX
+CURRENT_LIB
+CURRENT_NETLIST
+CURRENT_ODB
+CURRENT_POWERED_NETLIST
+CURRENT_SDC
+CURRENT_SDF
+CURRENT_SPEF
+CURRENT_STEP
+DEBUG
+DESIGN_CONFIG
+DESIGN_DIR
+ESTIMATE_PARASITICS
+EXIT_ON_ERROR
+EXT_NETLIST
+FLOW_FAILED
+GDS_INPUT
+GLB_CFG_FILE
+GND_NET
+GRT_CONGESTION_REPORT_FILE
+INSERT_BUFFER_COMMAND
+INSERT_BUFFER_COUNTER
+IO_READ_DEF
+LAST_TIMING_REPORT_TAG
+LEC_LHS_NETLIST
+LEC_RHS_NETLIST
+LOGS_DIR
+MAGIC_GDS
+MAGIC_SCRIPT
+MAX_METAL_LAYER
+MC_SDF_DIR
+MC_SPEF_DIR
+METAL_LAYER_NAMES
+OPENLANE_MOUNTED_SCRIPTS_VERSION
+OPENLANE_ROOT
+OPENLANE_VERSION
+OPENROAD_BIN
+PACKAGED_SCRIPT_0
+PDKPATH
+PROCESS_CORNER
+PWD
+RCX_DEF
+RCX_LEF
+RCX_LIB
+RCX_RULESET
+REPORTS_DIR
+REPORT_OUTPUT
+RESULTS_DIR
+RUN_DIR
+RUN_TAG
+SAVE_DEF
+SAVE_GDS
+SAVE_GUIDE
+SAVE_LIB
+SAVE_MAG
+SAVE_NETLIST
+SAVE_ODB
+SAVE_POWERED_NETLIST
+SAVE_SDC
+SAVE_SDF
+SAVE_SPEF
+SCRIPTS_DIR
+SCRIPT_DIR
+START_TIME
+STA_MULTICORNER
+STA_PRE_CTS
+SYNTH_EXPLORE
+SYNTH_SCRIPT
```

Comment:
Slate for removal- use OpenLane 2 or interactive flow if you want to custom-synth